### DISCUSSION TITLE
How to not use some block of code of teardown method after every testcase?

### DISCUSSION THREAD

**Discussion by poojapatel14710** [seleniumbase/SeleniumBase#1632] [Q&A] [answered]
Hello Sir,

I am using the Delete method of CRUD operation for deleting data from firebase in teardown method . But i don't want to delete that for every testcases, instead of that i want to delete for some testcases. So, what can i do?

The below block of code i added in teardown method. But i dont want to execute this after end of test case for some specific test case.

```
        # delete all connected channel after every testcase
        cred = credentials.Certificate('C:/Automation/SeleniumBase/demo/voltaic............json')
        default_app = firebase_admin.initialize_app(cred)
        db = firestore.client()
        firebase.FirebaseApplication('https://voltaic-.........firebaseio.com', None)
        accounts = db.collection('clients').document(".......xyz............").collection("accounts").get()
        for account in accounts:
            key = account.id
            db.collection('clients').document("..........xyz.............").collection("accounts").document(key).delete()
        firebase_admin.delete_app(default_app)
```

Thank you

**Comment by mdmintz** [comment, marked as answer]
Using this format: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_02 you can easily customize your ``tearDown()`` method. Set a variable in your tests that you will access in your ``tearDown()`` method to determine if you need to run custom code. (Eg. In a test: ``self.my_var = 1`` ... In the tearDown(): ``if self.my_var == 1:`` ...)

**Comment by poojapatel14710** [comment]
Okay, sir got it but if I set variable for example,

`self.delete_channel = True` in test case

then  i have to mention this variable  in every test case if i want to delete the channel and if I don't want to delete then need to mention 

`self.delete_channel = False`


is that right?

Because if i don't mention "self.delete_channel = False" in other test cases it throws an error
" if self.delete_channel == True:
E       AttributeError: 'TestUpdateTokenPage' object has no attribute 'delete_channel'"

**Reply by mdmintz** [reply]
That's what Python's ``hasattr`` is for.

```python
if hasattr(self, "delete_channel") and self.delete_channel:
    ...
```

**Comment by poojapatel14710** [comment]
got it . i set "self.delete_channel =False"  in tear up method, so default it takes False

**Reply by mdmintz** [reply]
You can use a custom ``setUp()`` method for initializing variables. Follow the example from: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_02
